kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoWindowedTable.java

```diff
@@ -457,13 +459,37 @@ public KeyValueIterator<WindowedKey, byte[]> backFetch(
   @Override
   public KeyValueIterator<WindowedKey, byte[]> fetchRange(
       final int kafkaPartition,
-      final Bytes fromKey,
-      final Bytes toKey,
+      final Bytes keyFrom,
+      final Bytes keyTo,
       final long timeFrom,
       final long timeTo
   ) {
-    throw new UnsupportedOperationException("fetchRange not yet supported for Mongo backends");
+    final List<KeyValueIterator<WindowedKey, byte[]>> segmentIterators = new LinkedList<>();
+    final var partitionSegments = kafkaPartitionToSegments.get(kafkaPartition);
+
+    for (final var segment : partitioner.range(kafkaPartition, timeFrom, timeTo)) {
+      final var segmentWindows = partitionSegments.segmentWindows.get(segment);
+
+      // Since we use a flat keyspace by concatenating the timestamp with the data key and have
+      // variable length data keys, it's impossible to request only valid data that's within
+      // the given bounds. Instead issue a broad request from the valid bounds and then post-filter
+      final var lowerBound = compositeKey(keyFrom, timeFrom);
+      final var upperBound = compositeKey(keyTo, timeTo);
 
+      final FindIterable<WindowDoc> fetchResults = segmentWindows.find(
+          Filters.and(
```

Comment:
I'm wondering if we should include these fields outside of the key and filter on them server-side. This has the potential to jack up our data transfer costs.